include/fiction/algorithms/simulation/sidb/operational_domain.hpp

```diff
@@ -794,6 +906,60 @@ class operational_domain_impl
         // if we made it here, the layout is operational
         return operational();
     }
+    /**
+     * Logs and returns the operational status at the given point `sp = (x, y)`. If the point has already been sampled,
+     * it returns the cached value. Otherwise, a ground state simulation is performed for all input combinations of the
+     * stored layout using the given simulation parameters. It terminates as soon as a non-operational state is found.
+     * In the worst case, the function performs \f$ 2^n \f$ simulations, where \f$ n \f$ is the number of inputs of the
+     * layout. This function is used by all operational domain computation techniques.
+     *
+     * Any investigated point is added to the stored `op_domain`, regardless of its operational status.
+     *
+     * @param sp Step point to be investigated.
+     * @return The operational status of the layout under the given simulation parameters.
+     */
+    operational_status is_step_point_suitable_for_given_cds(const step_point& sp, Lyt& lyt) noexcept
```

Comment:
In C++, function names should not contain phrases like `for_given_*` because that is obvious from the parameters it receives. This strategy might be necessary in languages that do not allow for function overloading.